packages/combobox/src/Combobox.tsx

```diff
@@ -1059,19 +1062,37 @@ export default function Combobox<M extends boolean>({
    * Global Event Handler
    *
    */
-  // Global backdrop click handler
-  const handleBackdropClick = ({ target }: MouseEvent) => {
-    const isChildFocused =
-      menuRef.current?.contains(target as Node) ||
-      comboboxRef.current?.contains(target as Node) ||
-      false;
 
-    if (!isChildFocused) {
-      setOpen(false);
+  /**
+   * We add two event handlers to the document to handle the backdrop click behavior.
+   * Intended behavior is to close the menu, and keep focus on the Combobox.
+   * No other click event handlers should fire on backdrop click
+   *
+   * 1. Mousedown event fires
+   * 2. We prevent `mousedown`'s default behavior, to prevent focus from being applied to the body (or other target)
+   * 3. Click event fires
+   * 4. We handle this event on _capture_, and stop propagation before the `click` event propagates all the way to any other element.
+   *  This ensures that even if we click on a button, that handler is not fired
+   * 5. Then we call `closeMenu`, setting `isOpen = false`, and rerender the component
+   */
+  useEventListener('mousedown', (mousedown: MouseEvent) => {
+    if (!doesComponentContainEventTarget(mousedown) && isOpen) {
+      mousedown.preventDefault(); // Prevent focus from being applied to body
```

Comment:
Also, @spark3-mdb @shaneeza would love for more than just myself to brainstorm ways this could possibly go wrong 😅 definitely nervous about this change breaking in some way.